lib/puppet_x/jenkins/provider/cli.rb

```diff
@@ -122,22 +133,31 @@ def self.cli(command, options = {})
       input = options.delete(:stdin)
     end
 
+    if options.key?(:tmpfile_as_param)
+      tmpfile_as_param = options[:tmpfile_as_param]
+    else
+      tmpfile_as_param = false
+    end
+
     Puppet.debug("#{sname} stdin:\n#{input}")
 
     # a tempfile block arg is not used to simplify mock testing :/
     tmp = Tempfile.open(sname)
     tmp.write input
     tmp.flush
     options[:stdinfile] = tmp.path
-    result = execute_with_retry(command, options)
+    FileUtils.chown 'jenkins', 'jenkins', tmp.path if tmpfile_as_param and File.exists?(tmp.path)
```

Comment:
We use umask 027. Then this file is created by root and not readable by jenkins.